Content.Client/Stylesheets/Redux/BaseStylesheet.Sheetlets.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Robust.Client.UserInterface;
+using Robust.Shared.Utility;
+
+namespace Content.Client.Stylesheets.Redux;
+
+public abstract partial class BaseStylesheet
+{
+    public StyleRule[] GetSheetletRules<TSheetTy>(Type sheetletTy)
+    {
+        Sheetlet<TSheetTy>? sheetlet = null;
+        try
+        {
+            if (sheetletTy.ContainsGenericParameters)
+            {
+                if (SandboxHelper.CreateInstance(sheetletTy.MakeGenericType(typeof(TSheetTy))) is Sheetlet<TSheetTy>
+                    sheetlet1)
+                    sheetlet = sheetlet1;
+            }
+            else if (SandboxHelper.CreateInstance(sheetletTy) is Sheetlet<TSheetTy> sheetlet2)
+            {
+                sheetlet = sheetlet2;
+            }
+        }
+        // thrown when `sheetletTy.MakeGenericType` is given a type that does not satisfy the type constraints of
+        // `sheetletTy`
+        catch (ArgumentException) { }
```

Comment:
yeah, this is the cleanest solution i've found without doing that. Tested a couple other methods and this ended up being the least slow. It's 170ms right now, so this probably should be fixed otherwise it could easily add a second or two to load times in the future.

I'm really not familliar with sandboxing / reflection, so a couple pointers would be helpful if I were to try and implement this.